### PR TITLE
update-device-activation-time

### PR DESCRIPTION
**Pull request by IloveJavaa**
Add some functions that returning the device activation time. 
Asana link - https://app.asana.com/0/0/1205072173934431/f

### CODE REVIEW

File: TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/Constants.kt [hunk 1]
```diff
@@ -35,10 +35,14 @@ object Constants {
     const val SHARED_PREF_CLICKID = "clickId"
     const val SHARED_PREF_DLV = "dlv"
     const val SHARED_PREF_PID = "pid"
+    const val SHARED_PREF_MESSAGE = "message"
```
Comment (Hemant-Mann) [inline TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/Constants.kt:38]: let's remove

File: TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/Util.kt [hunk 2]
```diff
@@ -314,6 +316,43 @@ object Util {
...
+    }
+    
+    private fun getBootTime(): String {
+        val bootTime = System.currentTimeMillis() - SystemClock.elapsedRealtime()
+        val deviceActivationTime = bootTime - SystemClock.uptimeMillis()
+        val activationDate = Date(deviceActivationTime)
+        return activationDate.toString()
+    }
+    
+    private fun getRootDir(): String {
```
Comment (Hemant-Mann) [inline TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/Util.kt:342]: osUpdateTime

File: TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/Util.kt [hunk 3]
```diff
@@ -314,6 +316,43 @@ object Util {
...
+        val systemDir = Environment.getRootDirectory()
+        val creationTime = systemDir.lastModified()
+        return Date(creationTime).toString()
+    }
+    
+    private fun getUptime(): String {
```
Comment (Hemant-Mann) [inline TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/Util.kt:348]: remove

**Review by Hemant-Mann — CHANGES_REQUESTED** [review]

File: TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/Util.kt (hunk 3 above)
Comment (IloveJavaa) [reply TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/Util.kt:348]: Dome

File: TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/Util.kt (hunk 2 above)
Comment (IloveJavaa) [reply TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/Util.kt:342]: Done

File: TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/Util.kt [hunk 4]
```diff
@@ -314,6 +316,35 @@ object Util {
...
+        var buildDate = ""
+        if (manufacturer.equals(Constants.DEVICE_BRAND)) {
+            buildDate = getSysPropertyPath(Constants.SYSTEM_PROPERTIES_DEVICE_BUILD_XIAOMI).toString()
+        } else {
+            buildDate = getSysPropertyPath(Constants.SYSTEM_PROPERTIES_DEVICE_BUILD_ANDROID).toString()
+        }
+        val activationTimestamp = buildDate!!.toLong() * 1000L
+        val activationDate = SimpleDateFormat("yyyy-MM-dd HH:mm:ss")
+            .format(Date(activationTimestamp))
+        println("Activation Date: $activationDate")
```
Comment (Hemant-Mann) [inline TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/Util.kt:330]:
```suggestion
```

File: TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/Util.kt [hunk 5]
```diff
@@ -314,6 +316,35 @@ object Util {
...
+        logger.info("Activation Date = $activationDate")
```
Comment (Hemant-Mann) [inline TrackierSDK/sdk-core/src/main/java/com/trackier/sdk/Util.kt:331]:
```suggestion
```

**Review by Hemant-Mann — APPROVED** [review]

**Pull request merged**